andi/andi.py

```diff
@@ -287,33 +303,49 @@ def _plan(class_or_func: Callable, *,
     else:
         arguments = inspect(class_or_func)
 
+    args_errs = defaultdict(list)  # type: Dict[str, List[str]]
+    non_injectable_errs = defaultdict(list)  # type: Dict[str, List[str]]
     for argname, types in arguments.items():
         sel_cls = _select_type(types, is_injectable, externally_provided)
+        init_str = ".__init__()" if is_class else ""
         if sel_cls is not None:
-            if sel_cls not in plan_seq:
-                plan = _plan(sel_cls,
+            if sel_cls not in plan_od:
+                plan, errors = _plan(sel_cls,
                              is_injectable=is_injectable,
                              externally_provided=externally_provided,
                              full_final_kwargs=True,
                              dependency_stack=dependency_stack)
-                plan_seq.update(plan)
-            type_for_arg[argname] = sel_cls
+                plan_od.update(plan)
+            if errors:
+                args_errs[argname].extend(errors)
+            else:
+                type_for_arg[argname] = sel_cls
         else:
-            if full_final_kwargs:
-                init_str = ".__init__()" if is_class else ""
-                if not types:
-                    msg = "Parameter '{}' is lacking annotations in " \
-                          "'{}{}'. Not possible to build a plan".format(
-                        argname, class_or_func, init_str)
-                else:
-                    msg = "Any of {} types are required ".format(types)
-                    msg += " for parameter '{}' ".format(argname)
-                    msg += " in '{}{}' but none can be provided".format(
-                        class_or_func, init_str)
-                raise NonProvidableError(msg)
-
-    plan_seq[class_or_func] = type_for_arg
-    return plan_seq
+            if not types:
+                msg = "Parameter '{}' is lacking annotations in " \
+                      "'{}{}'".format(
+                    argname, class_or_func, init_str)
+            else:
+                msg = "Any of {} types are required ".format(types)
+                msg += "for parameter '{}' ".format(argname)
+                msg += "in '{}{}' ".format(class_or_func, init_str)
+                msg += "but none of them is injectable or externally providable"
+            non_injectable_errs[argname].append(msg)
+
+    # Error managing
+    if full_final_kwargs:
+        args_errs.update(non_injectable_errs)
+    flatten_errors = [error
+                      for errors in args_errs.values()
+                      for error in errors]
+    if is_root_call and args_errs:
+        _raise_exception(class_or_func, args_errs)
+
+    plan_od[class_or_func] = type_for_arg
+    plan = Plan(plan_od.items())
+    plan._full_final_kwargs = not non_injectable_errs
```

Comment:
I think that's fine to remove the property, and just have a full_final_kwargs attribute without underscore.

Reply:
Done